actions/tamper_action.go

```diff
@@ -86,60 +89,637 @@ func ParseTamperAction(s *scanner.Scanner) (Action, error) {
+		case field == TCPFieldFlags:
+			gen.vUint = tcpFlagsToUint32(ta.NewValue)
+		case field < TCPFieldSrcPort:
+			// if field is an option, we need to convert the value to a byte slice
+			var b []byte
+			if val, err := strconv.ParseUint(ta.NewValue, 10, 64); err == nil {
+				b = make([]byte, 8)
+				binary.BigEndian.PutUint64(b, val)
+				b = b[8-tcpOptionLengths[field]:]
+			} else {
+				b = []byte(ta.NewValue)
+			}
+
+			gen.vBytes = b
+		case field == TCPLoad:
+			gen.vBytes = []byte(ta.NewValue)
+		default:
+			val, err := strconv.ParseUint(ta.NewValue, 10, 32)
+			if err != nil {
+				return nil, fmt.Errorf("invalid tamper rule: %q is not a valid value for field %q", ta.NewValue, ta.Field)
+			}
+
+			gen.vUint = uint32(val)
+		}
+
+		return &TCPTamperAction{
+			TamperAction: ta,
+			field:        field,
+			valueGen:     gen,
+		}, nil
+	}
+
+	return nil, fmt.Errorf("invalid tamper rule: %q is not a valid tamper mode for TCP", ta.Mode)
+}
+
+func (a *TCPTamperAction) Apply(packet gopacket.Packet) ([]gopacket.Packet, error) {
+	tcp := packet.Layer(layers.LayerTypeTCP).(*layers.TCP)
+	if tcp == nil {
+		return nil, fmt.Errorf("packet does not have a TCP layer")
+	}
 
-		return action, nil //nolint:nilerr
+	tamperTCP(tcp, a.field, a.valueGen)
+
+	// if tampering with TCP options, we need to update the data offset and checksum
+	if strings.HasPrefix(a.Field, "options") {
+		updateTCPDataOffAndChksum(tcp)
+		if ip := packet.Layer(layers.LayerTypeIPv4).(*layers.IPv4); ip != nil {
+			updateIPv4LengthAndChksum(ip)
+		}
 	}
 
-	if action.Action, err = ParseAction(s); err != nil {
-		if c, err2 := s.Peek(); err2 == nil && c == ')' {
-			action.Action = &SendAction{}
+	return a.Action.Apply(packet)
+}
+
+// tamperTCP modifies the given TCP field using the given value generator.
+func tamperTCP(tcp *layers.TCP, field TCPField, valueGen tamperValueGen) {
+	switch field {
+	case TCPFieldSrcPort:
+		tcp.SrcPort = layers.TCPPort(valueGen.uint(16))
+	case TCPFieldDstPort:
+		tcp.DstPort = layers.TCPPort(valueGen.uint(16))
+	case TCPFieldSeq:
+		tcp.Seq = valueGen.uint(32)
+	case TCPFieldAck:
+		tcp.Ack = valueGen.uint(32)
+	case TCPFieldDataOff:
+		tcp.DataOffset = uint8(valueGen.uint(8))
+	case TCPFieldWindow:
+		tcp.Window = uint16(valueGen.uint(16))
+	case TCPFieldUrgent:
+		tcp.Urgent = uint16(valueGen.uint(16))
+	case TCPFieldFlags:
+		setTCPFlags(tcp, uint16(valueGen.uint(16)))
+	case TCPLoad:
+		tcp.Payload = valueGen.bytes(1480)
+	default:
+		// find option in TCP header
+		var opt *layers.TCPOption
+		for i, o := range tcp.Options {
+			if field == TCPField(o.OptionType) {
+				opt = &tcp.Options[i]
+				break
+			}
+		}
+
+		// create option if it doesn't exist and move options-eol to the end of the list
+		if opt == nil {
+			tcp.Options = append(tcp.Options, layers.TCPOption{
+				OptionType: layers.TCPOptionKind(field),
+			})
+
+			ol := len(tcp.Options)
+			tcp.Options[ol-2], tcp.Options[ol-1] = tcp.Options[ol-1], tcp.Options[ol-2]
```

Comment:
The Geneva team just append new options to the list and I didn't see anywhere in scapy that it would automatically check / correct that. So I think it should be fine if we don't.